tests/unit/moduleapi/internalsecret.tcl

```diff
@@ -0,0 +1,275 @@
+        assert_equal {OK} [r internalauth.internalcommand]
+    }
+}
+
+start_cluster 1 0 [list config_lines $modules] {
+    set r [srv 0 client]
+
+    test {RM_Call of internal commands succeeds only for internal connections, and overrised user attachment} {
+        # Fail before authenticating as an internal connection.
+        assert_error {*unknown command*} {r internalauth.internal_rmcall internalauth.internalcommand}
+
+        # Authenticate as an internal connection.
+        set reply [r internalauth.getinternalsecret]
+        assert_equal {OK} [r auth "internal connection" $reply]
+
+        # Succeeds, since the user is overridden (otherwise would have failed
+        # since there is no user attached).
+        assert_equal {OK} [r internalauth.internal_rmcall_withuser internalauth.internalcommand]
+    }
+}
+
+start_cluster 1 0 [list config_lines $modules] {
+    set r [srv 0 client]
+
+    test {RM_Call with the `C` flag after setting thread-safe-context should fail} {
+        # New threadSafeContexts do not inherit the internal flag.
+        assert_error {*unknown command*} {r internalauth.internal_rmcall_detachedcontext internalauth.internalcommand}
+    }
+}
+
+start_cluster 1 0 [list config_lines $modules] {
+    set r [srv 0 client]
+
+    r config set appendonly yes
+    r config set appendfsync always
+    waitForBgrewriteaof r
+
+    test {AOF executes internal commands successfully} {
+        # Authenticate as an internal connection
+        set reply [r internalauth.getinternalsecret]
+        assert_equal {OK} [r auth "internal connection" $reply]
+
+        # Call an internal writing command
+        assert_equal {OK} [r internalauth.internal_rmcall_replicated set x 5]
+
+        # Reload the server from the AOF
+        r debug loadaof
+
+        # Check if the internal command was executed successfully
+        assert_equal {5} [r get x]
+    }
+}
+
+start_cluster 1 0 [list config_lines $modules] {
+    set r [srv 0 client]
+
+    test {Internal commands are not allowed from scripts} {
+        # Internal commands are not allowed from scripts
+        assert_error {*not allowed from script*} {r eval {redis.call('internalauth.internalcommand')} 0}
+
+        # Even after authenticating as an internal connection
+        set reply [r internalauth.getinternalsecret]
+        assert_equal {OK} [r auth "internal connection" $reply]
+        assert_error {*not allowed from script*} {r eval {redis.call('internalauth.internalcommand')} 0}
+
+        # Internal commands ARE shown in monitor output
+        set rd [redis_deferring_client]
+        $rd monitor
+        $rd read ; # Discard the OK
+        catch {r eval {redis.call('internalauth.internalcommand')} 0} err
+        assert_match "*not allowed from script*" $err
+        assert_match {*eval*internalauth.internalcommand*} [$rd read]
+        # No following log, since the command failed.
+        $rd close
```

Comment:
Not sure what you are testing here? That the eval command are shown in the monitor?

Reply:
Will remove